docs/service-guide/update-a-onepassword-secret.rst

```diff
@@ -0,0 +1,30 @@
+#####################################################
+Updating a secret stored in 1Password and VaultSecret
+#####################################################
+
+Secrets that are stored in 1Password are synchronized into Vault using the `installer/generate_secrets.py <https://github.com/lsst-sqre/phalanx/blob/master/installer/generate_secrets.py>`__ script.
+Once they are in Vault, they are accessible to the Vault Secrets Operator, which responds to creation of any ``VaultSecret`` resources in Kubernetes by grabbing the current value of the secret data in Vault.
+
+The Vault Secrets Operator doesn't independently detect if there are changes to data in Vault, though.
+It only provides a snapshot of the data in Vault at the time the ``VaultSecret`` was created.
```

Comment:
Maybe this is something we could fix instead of documenting this (or using this doc as a fall-back). In the vault-secrets-operator doc, they talk about two ways for secrets to get updated:

1. Use a `spec.version` field to specify the version of the Vault secret you want to use. I guess in that case we'd need to use the vault CLI or have a wrapper that reports on the different versions of a secret in vault.
2. Have vault-secrets-operator be configured to update secrets via the `VAULT_RECONCILIATION_TIME` environment variable configuration.

> The spec.type and spec.keys fields are handled in the same way for both versions of the KV secret engine. The spec.version field is only processed, when the secret is saved under a KVv2 secret engine. If you specified the VAULT_RECONCILIATION_TIME environment variable with a value greater than 0 every secret is reconciled after the given time. This means, when you do not specify spec.version, the Kubernetes secret will be automatically updated if the Vault secret changes.

I notice in [our configuration](https://github.com/lsst-sqre/phalanx/blob/master/services/vault-secrets-operator/values-base.yaml) we haven't set this, so I wonder if the value is zero?

@rra What do you think? Should we change vault-secrets-operator to automatically reconcile with Vault?

Reply:
Oh, I like this _much_ more. I would very much prefer to have `VAULT_RECONCILIATION_TIME=60` or something, and then document that it takes up to 60 seconds for changes to be reflected into Kubernetes Secrets.

Reply:
Yeah, let's set `VAULT_RECONCILIATION_TIME`. Spencer, do you want to add that to this diff? Looks like it's just a matter of adding `vault.reconciliationTime` to the `values-*.yaml` files for all the environments. 60 seconds is a bit aggressive, but hopefully the API call is cheap, so we can start there and change it if there's some problem.